iocore/eventsystem/UnixEventProcessor.cc

```diff
@@ -491,3 +489,36 @@ EventProcessor::spawn_thread(Continuation *cont, const char *thr_name, size_t st
 
   return e;
 }
+
+bool
+EventProcessor::has_tg_started(int etype)
+{
+  return thread_group[etype]._started == thread_group[etype]._count;
+}
+
+bool
+EventProcessor::has_tgs_started(std::vector<int> etypes)
+{
+  if (etypes.empty()) {
+    return false;
+  }
+  bool flag = true;
+  for (int etype : etypes) {
+    flag &= has_tg_started(etype);
+  }
+  return flag;
+}
+
+void
+thread_start_counter(EThread *t)
```

Comment:
That's a terrible name. Is "start" a noun or a verb? What about `check_and_signal_thread_group_start`?